main.go

```diff
@@ -81,18 +76,20 @@ func initLogs() {
 	}
 }
 
-func filterVulnerabilities(severityThresholdStr string, vulnerabilities []*clair.Vulnerability) []*clair.Vulnerability {
-	var ret []*clair.Vulnerability
+func filterVulnerabilities(severityThresholdStr string, vulnerabilities *grype_models.Document) *grype_models.Document{
+	var filtered []grype_models.Match
 
 	severityThreshold := vulutils.GetSeverityFromString(severityThresholdStr)
-	for _, vulnerability := range vulnerabilities {
-		if vulutils.GetSeverityFromString(vulnerability.Severity) < severityThreshold {
+	for _, vulnerability := range vulnerabilities.Matches {
```

Comment:
since `vulnerability` is a tmp copy It will be better to use `vulnerabilities.Matches[i]` instead.